### PR TITLE
add intial guidance on openshift start config

### PR DESCRIPTION
**Pull request by deads2k**
We've added configuration files that can describe how to launch openshift.  This is a very sparse introduction to their existence.  The actual contents of the config files is still changing pretty rapidly, so I haven't added details about the stanzas or their serializations.

@liggitt

### CODE REVIEW

File: using_openshift/master_node_configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+== Create the starting config files
+The `openshift start` command accepts flags that indicate that it should simply write the config file that it would have used and terminate.  This is useful for getting a starting point for the config.  You can do this by running `openshift start --writeconfig --master-config=master.yaml --node-config=node.yaml` or `openshift start master --write-config --config=master.yaml` or `openshift start node --write-config config=node.yaml`.
+
+== Use the config files
+Once you have modified the config file to your liking, you can make use of them by specifying them as an argument.  Keep in mind that if you specify a config file, *none of the other flags you pass in will be respected*.  You can run them like: `openshift start --master-config=master.yaml --node-config=node.yaml` or `openshift start master --config=master.yaml` or `openshift start node config=node.yaml`.
```
Comment (liggitt) [inline using_openshift/master_node_configuration.adoc]: "config files"

Comment (liggitt) [inline using_openshift/master_node_configuration.adoc]: bulletize the examples

File: using_openshift/master_node_configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+
+== Example master.yaml
+This is an example at a point in time.  You should run `--write-config` to generate a file of your own.
+
+[source]
+---
+apiVersion: v1
+assetConfig:
+  kubernetesPublicURL: https://10.13.137.161:8443
```
Comment (liggitt) [inline using_openshift/master_node_configuration.adoc]: exemplify ips?

File: using_openshift/master_node_configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+This is an example at a point in time.  You should run `--write-config` to generate a file of your own.
+
+[source]
+---
+allowDisabledDocker: true
+apiVersion: v1
+dnsDomain: local
+dnsIP: 10.13.137.161
+kind: NodeConfig
+masterKubeConfig: openshift.local.certificates/node-deads-dev-01/.kubeconfig
```
Comment (liggitt) [inline using_openshift/master_node_configuration.adoc]: exemplify hostname

**Comment by liggitt** [comment]
LGTM otherwise

File: using_openshift/master_node_configuration.adoc (hunk 2 above)
Comment (deads2k) [inline using_openshift/master_node_configuration.adoc]: > exemplify ips?

What private range do you like best?

**Comment by deads2k** [comment]
Comments addressed, except for the example ip.  Is there some private range we normal use other than  10.?

**Comment by liggitt** [comment]
Not sure. 10.0.0.1?

**Comment by deads2k** [comment]
comments addressed

**Comment by adellape** [comment]
@deads2k @liggitt Were you wanting to leave this open while config file work is ongoing? Or do you feel this is ready to merge now, and more can be added in a subsequent PR?

File: using_openshift/master_node_configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+The `openshift start` command accepts flags that indicate that it should simply write the config file that it would have used and terminate.  This is useful for getting a starting point for the config.  You can do this by running 
+*`openshift start --writeconfig --master-config=master.yaml --node-config=node.yaml`
+* `openshift start master --write-config --config=master.yaml`
+* `openshift start node --write-config config=node.yaml`
+
+== Use the config files
+Once you have modified the config files to your liking, you can make use of them by specifying them as an argument.  Keep in mind that if you specify a config file, *none of the other flags you pass in will be respected*.  You can run them like: 
+* `openshift start --master-config=master.yaml --node-config=node.yaml`
+* `openshift start master --config=master.yaml`
+* `openshift start node config=node.yaml`
```
Comment (sosiouxme) [inline using_openshift/master_node_configuration.adoc]: ITYM `openshift start node --config=node.yaml` (with the hyphens)

File: using_openshift/master_node_configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+kind: NodeConfig
+masterKubeConfig: openshift.local.certificates/node-hostname.example.org/.kubeconfig
+networkContainerImage: openshift/origin-pod:v0.4.1
+nodeName: hostname.example.org
+recordEvents: false
+servingInfo:
+  bindAddress: 0.0.0.0:10250
+  certFile: openshift.local.certificates/node-hostname.example.org/server.crt
+  clientCA: ""
+  keyFile: openshift.local.certificates/node-hostname.example.org/server.key
```
Comment (sosiouxme) [inline using_openshift/master_node_configuration.adoc]: When describing values, probably worth mentioning why these are not inlined... we want admins to be able to post these config files for help without having to redact sensitive stuff (urls and IPs may still be sensitive, but externalizing those seems excessive).

Just an observation, no need to hold up the PR

**Comment by liggitt** [comment]
@adellape merge it... I have info that will need to go on top of it for etcd/node https config

**Comment by liggitt** [comment]
@adellape, luke's comments are correct if you want to add those tweaks

**Comment by adellape** [comment]
@liggitt Ack, thx.

@sosiouxme Thx, will address in a follow-up PR.

**Pull request merged**